lib/utf8.js

```diff
@@ -0,0 +1,12 @@
+"use strict";
+
+module.exports.decode = function (input) {
+  if (Buffer.isBuffer(input) === false) {
```

Comment:
Ah, that makes more sense! In that case the code is just confusing, and a separate function decodeArrayOfBytes would be good.